src/poll/GenericInput.js

```diff
@@ -0,0 +1,72 @@
+import React, {Component} from 'react'
+import {pickBy} from 'lodash'
+
+class GenericInput extends Component{
+
+  state = {
+    charsLeft: this.props.maxLength || 0,
+    hasFocus: false,
+  }
+
+  render(){
+    return(
+      <span>
+        <label htmlFor={this.props.id}>
+        {this.props.label} 
+        &nbsp;
+        {this.state.hasFocus &&
+          `(${this.state.charsLeft})`
```

Comment:
It might actually be simpler to just derive the number of characters left, as you've already done, but without storing it in the state, and just render the value directly.

```js
render() {
  ...
  <input
    ref={(input) => this.input = input}
    ...
  />
  ...
}
```

```js
cont charsLeft = {this.props.maxLength - this.input.value.length}
```